src/lib/deploy.js

```diff
@@ -180,50 +180,70 @@ async function deploy({ network, yes }) {
   let zkAppPrivateKey = PrivateKey.fromBase58(privateKey); //  The private key of the zkApp
   let zkAppAddress = zkAppPrivateKey.toPublicKey(); //  The public key of the zkApp
 
-  let verificationKey = await step(
+  const verificationKey = await step(
     'Generate verification key (takes 1-2 min)',
     async () => {
       let { verificationKey } = await zkApp.compile(zkAppAddress);
       return verificationKey;
     }
   );
 
-  // Get the transaction fee amount to deploy specified by the user
-  let response = await prompt({
-    type: 'input',
-    name: 'fee',
-    message: (state) => {
-      const style = state.submitted && !state.cancelled ? green : reset;
-      return style('Set transaction fee to deploy (in MINA):');
-    },
-    validate: (val) => {
-      if (!val) return red('Fee is required.');
-      if (isNaN(val)) return red('Fee must be a number.');
-      return true;
-    },
-    result: (val) => val.trim().replace(/ /, ''),
-  });
-
-  let { fee } = response;
-  if (!fee) return;
+  let fee;
+  if (yes) {
+    // If running in non-interactive mode, get the transaction fee amount from the user's config file
+    fee = config.networks[network]?.fee;
+  } else {
+    // If running in interactive mode, get the transaction fee amount from the user's input
+    let feeResponse = await prompt({
+      type: 'input',
+      name: 'fee',
+      message: (state) => {
+        const style = state.submitted && !state.cancelled ? green : reset;
+        return style('Set transaction fee to deploy (in MINA):');
+      },
+      validate: (val) => {
+        if (!val) return red('Fee is required.');
```

Comment:
Are there any other validation cases or warnings we can think of to give the user more guard rails (ie no negative fees or maybe a warning if the fee is set to large to give the user another sanity check)?

Reply:
Good catch! I added the case for negative values but I'm not sure what to do for a large fee amount. Should have more discussion on that.

Reply:
Nice! I don't have an answer on what to do if the fee input is a large amount but I wanted to mention it. I think we should discuss it further. In general, I'm for helping the users prevent common mistakes with warnings if possible.

Reply:
+1 would be good discussion, but not urgent. 

I almost suggested a sanity check like <=1 MINA in response to Yoni's good suggestion, but imagine there is a network spam attack and a dev wants to get an update deployed at any cost, they might want to raise it. IOW, maybe such a sanity check would better be implemented as an additional step check, asking the user to confirm b/c it looks high, rather than blocking them from proceeding as validation would do.